### ISSUE TITLE
Function and Property names encryption

### ISSUE THREAD

**Issue by kodeine** [naneau/php-obfuscator#22]
Hello,
Is there a way to encrypt function names and property names?
Thank you for hard work on this library.

for example this is the output for your review.
```php
    public function setTimeout(int $spb436b4)
    {
        $this->config->timeout = $spb436b4;
    }
```

**Comment by ipso** [comment]
Yes, that would be a really nice feature. Encrypting strings would be wonderful too!

**Comment by naneau** [comment]
This library was originally written so that all external (`public`, `protected`) API would not be changed. That way you can distribute your code as a library, but have its code be hard to read/modify.

Obfuscation of public methods won't work, because you'd have to find all places where they are called and replace them as well. This means that the scanning would have to expand to all methods calls. I can look into it, but it'll likely be more error-prone than internal (`private`) encryption.

**Comment by kodeine** [comment]
I think we can define project path and then it can find replace all methods and strings. Encrypting strings would make this package really unique and intuitive.

**Comment by ipso** [comment]
In our case we only want to obfuscate a single class file, which has about 2-3 public functions and the rest are all for "internal" use only. So we would like to obfuscate all but those public functions just within the class itself. 

Several of the strings (ie: private $myvar = "secretkey"; ) would be nice to obfuscate too just for added protection.

**Comment by gburtini** [comment]
See the last few comments of #28 for my start on a solution to this.

**Comment by naneau** [comment]
Closing for #28
